command-time.plugin.zsh

```diff
@@ -3,36 +3,70 @@ _command_time_preexec() {
   if [ -n "$ZSH_COMMAND_TIME_EXCLUDE" ]; then
     cmd="$1"
     for exc ($ZSH_COMMAND_TIME_EXCLUDE) do;
-      if [ "$(echo $cmd | grep -c "$exc")" -gt 0 ]; then
+      if grep -qF "$exc" <<< "$cmd"; then
         # echo "command excluded: $exc"
         return
       fi
     done
   fi
 
-  timer=${timer:-$SECONDS}
+  _command_time_timer=${timer:-$SECONDS}
   ZSH_COMMAND_TIME_MSG=${ZSH_COMMAND_TIME_MSG-"Time: %s"}
   ZSH_COMMAND_TIME_COLOR=${ZSH_COMMAND_TIME_COLOR-"white"}
+  ZSH_COMMAND_TIME_NO_COLOR=${ZSH_COMMAND_TIME_NO_COLOR-""}
   export ZSH_COMMAND_TIME=""
 }
 
 _command_time_precmd() {
-  if [ $timer ]; then
-    timer_show=$(($SECONDS - $timer))
-    if [ -n "$TTY" ] && [ $timer_show -ge ${ZSH_COMMAND_TIME_MIN_SECONDS:-3} ]; then
+  local timer_show
+  if [ $_command_time_timer ]; then
+    timer_show=$(($SECONDS - $_command_time_timer))
+    if [ -n "$TTY" ] && [[ $timer_show -ge ${ZSH_COMMAND_TIME_MIN_SECONDS:-3} ]]; then
       export ZSH_COMMAND_TIME="$timer_show"
       if [ ! -z ${ZSH_COMMAND_TIME_MSG} ]; then
         zsh_command_time
       fi
     fi
-    unset timer
+    unset _command_time_timer
   fi
 }
 
 zsh_command_time() {
+  local days hours min sec sec_fmt timer_show color
   if [ -n "$ZSH_COMMAND_TIME" ]; then
-    timer_show=$(printf '%dh:%02dm:%02ds\n' $(($ZSH_COMMAND_TIME/3600)) $(($ZSH_COMMAND_TIME%3600/60)) $(($ZSH_COMMAND_TIME%60)))
-    print -P "%F{$ZSH_COMMAND_TIME_COLOR}$(printf "${ZSH_COMMAND_TIME_MSG}\n" "$timer_show")%f"
+    # Round to integers
+    days=$(( ${ZSH_COMMAND_TIME%.*} / 86400 ))
+    hours=$(( ${ZSH_COMMAND_TIME%.*} / 3600 % 24 ))
+    min=$(( ${ZSH_COMMAND_TIME%.*} / 60 % 60 ))
```

Comment:
Maybe, let's use the `zsh` feature 'typeset -i` here? Then it will be unnecessary to use bashism with string operations.

It's a bit more obvious what's going on without string substitution
```suggestion
    typeset -i days=$(( ${ZSH_COMMAND_TIME} / 86400 ))
    typeset -i hours=$(( ${ZSH_COMMAND_TIME} / 3600 % 24 ))
    typeset -i min=$(( ${ZSH_COMMAND_TIME} / 60 % 60 ))
```